update design (#179)
Microsoft/agogosml: docs/DESIGN.rst

and focus on building the business logic inside the app container while
implementing a loosely coupled services approach.

> - Input container - Receive events from Azure Event Hub / Kafka.

Reviewer: - Input Container - Receive events from Azure Event Hub / Kafka.

(capitalize Container)

> - Init Container - Runs on load on the same pod as the customer application. Init container loads the ML module to a temporary storage (stored on Azure Blob Storage) which is then accessible by the Customer Application. This enables the Customer App to have access to the ML module while being oblivious to the location/implementation/management of it.

Reviewer 2: ```suggestion
- Init Container - Loads artifacts before initializing the customer application. The Init container (Artifact Loader)  loads the ML module to a temporary storage which is then accessible to the Customer Application.
```

- Customer App - Models implemented in PySpark, Tensorflow, scikit-learn and R.

> - Output container - send the results of the models to Azure Event Hub / Kafka or other data source.

Reviewer: Same as above
